odoo/addons/base/models/ir_fields.py

```diff
@@ -439,6 +439,12 @@ def _str_to_many2many(self, model, field, value):
 
     @api.model
     def _str_to_one2many(self, model, field, records):
+        name_create_enabled_fields = self._context.get('name_create_enabled_fields') or {}
+        relative_name_create_enabled_fields = {}
+        for key in name_create_enabled_fields:
+            prefix = field.name + '/'
+            if key.startswith(prefix):
+                relative_name_create_enabled_fields[key[len(prefix):]] = name_create_enabled_fields[key]
```

Comment:
`field` is a constant so `prefix` is a constant, so it should be possible to lift outside the loop, and also might be clearer as a listcomp? e.g.
```suggestion
        prefix = field.name + '/'
        relative_name_create_enabled_fields = {
            k[len(prefix):]: v
            for k, v in name_create_enabled_fields.items()
            if key.startswith(prefix)
        }
```